crates/kubelet/src/state/common/mod.rs

```diff
@@ -73,11 +75,11 @@ pub trait GenericProvider: 'static + Send + Sync {
     /// The state of the provider itself.
     type ProviderState: GenericProviderState;
     /// The state that is passed between Pod state handlers.
-    type PodState: GenericPodState;
+    type PodState: GenericPodState + ResourceState<SharedState = Self::ProviderState>;
```

Comment:
Again I kind of feel like this is changing the design of one component in a PR that is concerned with another component.  But I guess that it emerges from a need to model shared state for containers.

Reply:
This was a little weird because the shared state type does need to be specified in `ResourceState`, but it was also needed on `Provider` for ergonomics. This line was changed to enforce that both of those types were the same.